assets/js/hocs/with-feedback-prompt/index.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Fragment } from '@wordpress/element';
+import { InspectorControls } from '@wordpress/block-editor';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import FeedbackPrompt from './feedback-prompt.js';
+
+const blocksFeedback = {
+	'woocommerce/cart': __(
+		'We are currently working on improving our checkout and providing merchants with tools and options to customize their checkout to their stores needs.',
+		'woo-gutenberg-products-block'
+	),
+	'woocommerce/checkout': __(
+		'We are currently working on improving our checkout and providing merchants with tools and options to customize their checkout to their stores needs.',
+		'woo-gutenberg-products-block'
+	),
+};
+
+/**
+ * Adds a feedback prompt to the editor sidebar.
+ *
+ * @param {WPComponent} BlockEdit Original component.
+ *
+ * @return {WPComponent} Wrapped component.
+ */
+const withFeedbackPrompt = createHigherOrderComponent( ( BlockEdit ) => {
+	return ( props ) => {
+		const feedbackPromptText = blocksFeedback[ props.name ];
```

Comment:
This is a neat approach but I'm wary of having all the customization for feedback prompt in this one file.  It'd be more discoverable and easier to work with if individual blocks defined their specific elements.  What about one of these options?

### Option 1: specific named props for feedback prompt content and link.

In this option, edit components for blocks would be fed specific props that get picked up automatically by this HOC.  They'd need to be namespaced appropriately.

- `wcFeedbackContent`: For the custom content.
- `wcFeedbackUrl`: For the url where people submit their feedback.

### Option 2: use React.Context

This would be similar to option 2 but would utilize react context instead and the `withFeedbackPrompt` hoc would just read the values from the context.  Edit blocks would then just explicitly wrap anything receiving feedback with the feedback prompt provider:

_edit_
```jsx
<FeedbackProvider value={ feedBackConfig }>
    <EditComponent>
</FeedbackProvider>
```

_withFeedbackPrompt_:

```jsx
const withFeedbackPrompt = createHigherOrderComponent( ( BlockEdit ) => {
	return ( props ) => {
        const { content, url } = useFeedbackContext();
        if ( content ) { 
            /** do stuff **/
```

On the surface it seems like the context would be a bit more work, but the advantage is that the `EditComponent` or anything else in its composition doesn't have to care or worry about passing through any of the props and I like the explicitness in using the provider which makes it clear that this is a component with a feedback element.  I wonder if the Provider itself could take care of adding the `editor.BlockEdit` filter so everything is contained?

Reply:
I was playing with this idea but I'm not sure it's possible. The `editor.BlockEdit` is applied on an upper level than the block `edit` function is called, so wrapping the `<Edit>` block with a context provider, doesn't make that context available when calling the `editor.BlockEdit` filter.

I pushed a WIP for the checkout block here: https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1356/commits/21f889b021ceea6fef722efab9663799829bc769. Something similar would happen with props (option 1): props passed in the block `edit` function are not available in the `editor.BlockEdit`.

Please, let me know if I'm missing something!

***

To solve the issue you raised:

> The only thing I'm wary about is having all the customization for feedback prompt in this one file (which is right now just content). It'd be more discoverable and easier to work with if individual blocks defined their specific elements. 

If the options above don't work, I thought about some alternative solutions:
* If we don't use the filter but rely only on the HOC like in 718d3f2cc0258be5c767866b61a5cbb24aea2555, we would have access to props/context defined in the `edit` function.
* We could alternatively create a `blockFeedbackRegistry` similar to the `blocksRegistry` we currently have in place. That would allow us to call a `registerBlockFeedback()` function right after `registerBlockType()` and have access to the `blockFeedbackRegistry` inside the `editor.BlockEdit` filter.

Thoughts?